addons/website/static/src/js/editor/snippets.options.js

```diff
@@ -2822,6 +2822,165 @@ options.registry.ReplaceImage = options.Class.extend({
+        await this._super(...arguments);
+        const styleEl = document.getElementById('conditional_visibility');
+        if (styleEl) {
+            styleEl.parentNode.removeChild(styleEl);
+        }
+
+        this._userValueWidgets.forEach(async widget => {
+            let ids = this.$target[0].dataset[widget._methodsParams.attributeName + 'Id'];
+            if (!ids || !widget.m2oModel) {
+                widget._value = "[]";
+                return;
+            }
+            ids = ids.split(',').map(id => parseInt(id));
+            const selectedRecords = await this._rpc({
+                model: widget.m2oModel,
+                method: 'read',
+                args: [ids, ['display_name']],
+            });
+            widget._value = JSON.stringify(selectedRecords);
+
+            const validIds = selectedRecords.map(record => record.id);
+            const rottenIds = ids.filter(id => !validIds.includes(id));
+            if (rottenIds) {
+                this.selectValue(false, widget._value, widget._methodsParams);
+            }
+        });
+    },
+
+    /**
+     * Handles inserting the IDs and values in data-attribues
+     * @see this.selectClass
+     */
+    selectValue(previewMode, widgetValue, params) {
+        const recordData = JSON.parse(widgetValue);
+        const ids = [];
+        const values = [];
+        for (const record of recordData) {
+            ids.push(record.id);
+            values.push(record[params.callWith]);
+        }
+        if (ids.length && values.length) {
+            this.$target[0].dataset[`${params.attributeName}Id`] = ids.join(',');
+            this.$target[0].dataset[params.attributeName] = values.join(',');
+        } else {
+            delete this.$target[0].dataset[params.attributeName];
+            delete this.$target[0].dataset[`${params.attributeName}Id`];
+        }
+    },
+
+    /**
+     * extends data attribute to open the toggler when conditionnal is selected
+     * @override
+     */
+    selectDataAttribute(previewMode, widgetValue, params) {
+        this._super(...arguments);
+        if (params.attributeName === 'visibility' && widgetValue === 'conditional') {
+            const collapseEl = this.$el.children('we-collapse')[0];
+            this._toggleCollapseEl(collapseEl);
+        }
+    },
+    /**
+     * @override
+     */
+    async onTargetHide() {
+        this.$target[0].previousStyle = this.$target[0].style;
+        this.$target[0].style.display = 'none';
+    },
+    /**
+     * @override
+     */
+    async onTargetShow() {
+        this.$target[0].style = this.$target[0].previousStyle;
+    },
+
+
+    /**
+     * Reades the attributes of the traget and creates the selectors
+     * Store them in data-attributes to then be reapplied by content/inject_dom.js
+     * @override
+    */
+    async cleanForSave() {
+        this.$target[0].style = this.$target[0].previousStyle;
+        delete this.$target[0].dataset.invisible;
```

Comment:
`dataset.invisible` is for elements which should always be invisible if I remember well, so I don't think there should even be one here?